From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lunatickochiya <125438787@qq.com>
Date: Sat, 3 Oct 2026 22:00:00 +0800
Subject: [PATCH] strace: use bundled kernel headers

Build against the bundled kernel headers instead of using the current
kernel ones. This ensures strace is using the kernel headers it is
written against, and not a random one that may contain breaking uapi
changes (which happen from time to time).

Backport of upstream OpenWrt commit e7873c4d ("strace: use bundled
kernel headers") for strace 6.12 (openwrt-24.10 based trees).

Signed-off-by: lunatickochiya <125438787@qq.com>

--- a/package/devel/strace/Makefile
+++ b/package/devel/strace/Makefile
@@ -10,7 +10,7 @@
 
 PKG_NAME:=strace
 PKG_VERSION:=6.12
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=https://strace.io/files/$(PKG_VERSION)
@@ -31,8 +31,6 @@
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/nls.mk
 
-HOST_CFLAGS += -I$(LINUX_DIR)/user_headers/include
-
 CONFIGURE_VARS+= \
 	LDFLAGS_FOR_BUILD="$(HOST_LDFLAGS)" \
 	CPPFLAGS_FOR_BUILD="$(HOST_CPPFLAGS)" \
@@ -72,6 +70,7 @@
 	--with-libdw=$(if $(CONFIG_STRACE_LIBDW),yes,no) \
 	--with-libunwind=$(if $(CONFIG_STRACE_LIBUNWIND),yes,no) \
 	--enable-mpers=no \
+	--enable-bundled=yes \
 	--without-libselinux
 
 MAKE_FLAGS := \
